nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -135,8 +135,8 @@
     private volatile Set<Future<?>> loggingFutures = new HashSet<>(2);
     private final NotificationServiceManager serviceManager;
 
-    public RunNiFi(final File bootstrapConfigFile, final boolean verbose) throws IOException {
```

Comment:
I do not recommend changing the signature of a public constructor.  Although there is no other use of the RunNiFi constructor in the NiFi project, someone might be using it in their own code, elsewhere.  Is this essential to your testability goal?  It does look like we're not sure about if/how to use `verbose`, but that might be a different ticket.  If it is essential, how about a second constructor?

With respect to the `File bootstrapConfigFile` argument, I understand there is a difference because you make `getBootstrapConfigFile()` an instance method, so RunNiFi will not separately call `getBootstrapConfigFile()` first.  What if we allowed that as a nullable argument, where it is accepted if given, and looked it up if `null` is passed in?  Again, I think this would preserve compatibility with possible existing callers.